src/shadowbox/server/manager_service.ts

```diff
@@ -302,32 +318,58 @@ export class ShadowsocksManagerService {
   public async setAccessKeyDataLimit(req: RequestType, res: ResponseType, next: restify.Next) {
     try {
       logging.debug(`setAccessKeyDataLimit request ${JSON.stringify(req.params)}`);
-      const limit = req.params.limit as DataLimit;
-      if (!limit) {
-        return next(new restifyErrors.MissingParameterError(
-            {statusCode: 400}, 'Missing `limit` parameter'));
-      } else if (!Number.isInteger(limit.bytes)) {
-        return next(new restifyErrors.InvalidArgumentError(
-            {statusCode: 400}, '`limit` must be an integer'));
+      const accessKeyId = validateAccessKeyId(req.params.id);
+      const limit = validateDataLimit(req.params.limit);
+      await this.accessKeys.setAccessKeyDataLimit(accessKeyId, limit);
```

Comment:
Are you awaiting for enforcement to happen? It may be more efficient not to wait - you already know that settting the limit succeeded synchronously.